### ISSUE TITLE
apt update breaks avocode.desktop

### ISSUE THREAD

**Issue by therealkevinard** [avocode/avocode#1480]
* **Version**:  avocode:amd64 2.14.3
* **Web/Desktop**: Desktop
* **Platform**:  Ubuntu 16.04 xenial
  - detailed system info: 
	```
	 OS: Ubuntu 16.04 xenial
	 Kernel: x86_64 Linux 4.4.0-79-generic
	 Shell: bash 4.3.48
	 DE: GNOME
	 WM: GNOME Shell
	 GTK Theme: Numix-Dark [GTK2/3]
	 CPU: Intel Core i7-4790 @ 8x 4GHz [27.8°C]
	 GPU: Mesa DRI Intel(R) Haswell Desktop
	```
  - apt repo: 
	```
	 deb http://linux.avocode.com/apt all main
	```

Using the repository install, every update replaces my avocode.desktop launcher file (`/usr/share/applications/avocode.desktop`) with a non-functional launcher that the system ignores. 
So: With every update I have to manually rebuild my avocode.desktop launcher from scratch and reload my desktop. 
It's pretty trivial, but still - kind of a nuisance, I guess. I'd imagine there are other users out there who don't know how to handle the situation and just chalk it up to "Avocode is broken. That app is stupid" :cry: 

**The launcher updating leaves me with:** 
```
 [Desktop Entry]
 Name=Avocode
 Comment=The bridge between designers and developers
 Exec=./resources/Avocode
 Terminal=false
 Type=Application
 Icon=./resources/linux/icons/48.png
 GenericName=Design Inspector
 StartupNotify=true
 Categories=GNOME;GTK;Utility;Development;
 MimeType=text/plain;
```
**The patch I run on every update**
```
 [Desktop Entry]
 Exec=avocode
 Icon=/opt/Avocode/avocode.png
```

**Relevant log lines from this morning's updates:** 
- syslog
```
 -- syslog entries (filtered: everything to-do with avocode.desktop)
 Jun 15 08:57:05 kevin-Inspiron-3847 gnome-session[2119]: (gnome-software:2299): As-WARNING **: failed to rescan: Failed to parse /usr/share/applications/avocode.desktop.dpkg-new file: cannot process file of type text/plain
 Jun 15 08:57:05 kevin-Inspiron-3847 gnome-session[2119]: (gnome-software:2299): As-WARNING **: failed to rescan: Failed to parse /usr/share/applications/avocode.desktop.dpkg-tmp file: cannot process file of type text/plain
 Jun 15 08:57:09 kevin-Inspiron-3847 gnome-session[2119]: (gnome-software:2299): As-WARNING **: failed to rescan: Failed to parse /usr/share/applications/avocode.desktop file: cannot process file of type application/x-desktop
```
- dpkg
```
-- dpkg.log (filtered: everything to-do with avocode)
 2017-06-15 08:57:04 upgrade avocode:amd64 2.14.2 2.14.3
 2017-06-15 08:57:04 status half-configured avocode:amd64 2.14.2
 2017-06-15 08:57:04 status unpacked avocode:amd64 2.14.2
 2017-06-15 08:57:04 status half-installed avocode:amd64 2.14.2
 2017-06-15 08:57:05 status triggers-pending hicolor-icon-theme:all 0.15-0ubuntu1
 2017-06-15 08:57:05 status half-installed avocode:amd64 2.14.2
 2017-06-15 08:57:05 status triggers-pending desktop-file-utils:amd64 0.22-1ubuntu5.1
 2017-06-15 08:57:05 status half-installed avocode:amd64 2.14.2
 2017-06-15 08:57:05 status triggers-pending gnome-menus:amd64 3.13.3-6ubuntu5.1~xenial
 2017-06-15 08:57:05 status triggers-pending mime-support:all 3.59ubuntu1
 2017-06-15 08:57:09 status half-installed avocode:amd64 2.14.2
 2017-06-15 08:57:10 status unpacked avocode:amd64 2.14.3
 2017-06-15 08:57:10 status unpacked avocode:amd64 2.14.3
 2017-06-15 08:57:13 configure avocode:amd64 2.14.3 <none>
 2017-06-15 08:57:13 status unpacked avocode:amd64 2.14.3
 2017-06-15 08:57:13 status half-configured avocode:amd64 2.14.3
 2017-06-15 08:57:13 status installed avocode:amd64 2.14.3
```

**Comment by jakubzitny** [comment]
Hey @therealkevinard, sorry you have this problem, we already know about this, and we'll fix it upcoming weeks. 😉

**Comment by therealkevinard** [comment]
I figured you did but I didn't see anything logged here, so.... y'know... covering bases and all.  :smile:

**Comment by jakubzitny** [comment]
Hey @therealkevinard, can you please try installing this bundle if it works for you properly? It's 64-bit .deb -> https://upx.cz/ibg.

Install by `dpkg -i path/to/avocode_2.14.4-beta.1_amd64.deb`

**Comment by therealkevinard** [comment]
**tldr;**
**Muy Bien!** :smiley: The app itself fails here (just shows the chat window, but the rest of init fails), but the launcher part we're worried about seems golden. 

**The long-winded, logg-y version** (guess I'm just in one of those moods today?): 
1. removed the repo version, reinstalled from apt 
```
kevin@kevin-Inspiron-3847:/opt$ sudo apt-get install avocode 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  avocode
0 upgraded, 1 newly installed, 0 to remove and 2 not upgraded.
Need to get 0 B/48.1 MB of archives.
After this operation, 190 MB of additional disk space will be used.
Selecting previously unselected package avocode.
(Reading database ... 398749 files and directories currently installed.)
Preparing to unpack .../avocode_2.14.3_amd64.deb ...
Unpacking avocode (2.14.3) ...
Processing triggers for hicolor-icon-theme (0.15-0ubuntu1) ...
Processing triggers for desktop-file-utils (0.22-1ubuntu5.1) ...
Processing triggers for gnome-menus (3.13.3-6ubuntu5.1~xenial) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Setting up avocode (2.14.3) ...
```

2. yep. installed. 
```
kevin@kevin-Inspiron-3847:/opt$ which avocode 
/usr/local/bin/avocode
```

3. customize the apt avocode.desktop
```
// mostly just this line, as a flag 
Comment=The bridge between designers and developers - custom launcher
```

4. squash the one you sent over the repo version. 
```
kevin@kevin-Inspiron-3847:/opt$ sudo dpkg -i /home/kevin/Downloads/avocode/avocode_2.14.4-beta.1_amd64.deb 
(Reading database ... 398836 files and directories currently installed.)
Preparing to unpack .../avocode_2.14.4-beta.1_amd64.deb ...
Unpacking avocode (2.14.4-beta.1) over (2.14.3) ...
Setting up avocode (2.14.4-beta.1) ...
Processing triggers for desktop-file-utils (0.22-1ubuntu5.1) ...
Processing triggers for gnome-menus (3.13.3-6ubuntu5.1~xenial) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.15-0ubuntu1) ...
```

5. check avocode.destop
```
// just showing the important parts 
[Desktop Entry]
Comment=The bridge between designers and developers // <-- updated 
Exec=avocode // <-- good! 
Icon=/opt/Avocode/avocode.png // <-- good! 

```

6. Tried from launcher: Good! Launched the beta app, and I didn't have to do the `ctrl+f2... r... wait...` thing.

**Comment by jakubzitny** [comment]
Hey, we've just released another minor version of Avocode — `2.14.4`, I've added the changes from the beta, so the installation should now end with correct and working shortcut and icon in Ubuntu.

Can you confirm that it works like that, please?

**Comment by jakubzitny** [comment]
Hey, we've just released another minor version of Avocode — 2.14.4, I've added the changes from the beta, so the installation should now end with correct and working shortcut and icon in Ubuntu.

Can you confirm that it works like that, please?

**Comment by therealkevinard** [comment]
Flawless! Thank you, sir!
